### DISCUSSION TITLE
Souffle Early Stopping

### DISCUSSION THREAD

**Discussion by philzook58** [souffle-lang/souffle#2072] [Q&A]
I'm interested in using souffle for problems where a counter or an external function may keep creating new unseen values so that datalog may in principle not terminate.
I was wondering if there was a way to invoke souffle such that it would stop after N iterations, some timeout, or some database size and still dump the current database seen so far?

**Comment by b-scholz** [comment]
We have something like this; can you look up the `.limitsize` directive.
https://souffle-lang.github.io/directives
